### PR TITLE
[Hotfix] Normalize embedding

### PR DESCRIPTION
**Pull request by Eleven1Liu**
**Description** 
(1) Normalize word embedding for both torchtext vectors and vectors from the `embed_file`.
(2) Rename `--normalize` to `--normalize_embed`.

### CODE REVIEW

File: libmultilabel/data_utils.py [hunk 1]
```diff
@@ -135,8 +136,9 @@ def load_or_build_text_dict(
         embed_file (str): Path to a file holding pre-trained embeddings.
         embed_cache_dir (str, optional): Path to a directory for storing cached embeddings. Defaults to None.
         silent (bool, optional): Enable silent mode. Defaults to False.
-        normalize (bool, optional): Whether the word embeddings divide by `float(np.linalg.norm(vector) + 1e-6)`.
-        Defaults to False.
+        normalize_embed (bool, optional): Whether the word embeddings divide by `float(np.linalg.norm(vector) + 1e-6)`.
```
Comment (cjlin1) [inline libmultilabel/data_utils.py:139]: please check my comment on the earlier PR about the writing of this..

File: libmultilabel/data_utils.py [hunk 2]
```diff
@@ -135,8 +136,9 @@ def load_or_build_text_dict(
...
+            Defaults to False. We follow the normalization method here:
```
Comment (cjlin1) [inline libmultilabel/data_utils.py:140]: I think this is the normalization method everyone uses? (x/|x|)  so why "follow ..." here?
What I would write is: We use the constant 1e-6 by following http://.. for an internal experiment of reproducing their results.

File: libmultilabel/data_utils.py [hunk 3]
```diff
@@ -158,15 +160,21 @@ def load_or_build_text_dict(
 
     if os.path.exists(embed_file):
         logging.info(f'Load pretrained embedding from file: {embed_file}.')
-        embedding_weights = get_embedding_weights_from_file(vocabs, embed_file, silent, normalize)
-        vocabs.set_vectors(vocabs.stoi, embedding_weights,
-                           dim=embedding_weights.shape[1], unk_init=False)
+        embedding_weights = get_embedding_weights_from_file(vocabs, embed_file, silent)
+        vocabs.set_vectors(vocabs.stoi, embedding_weights, dim=embedding_weights.shape[1])
```
Comment (cjlin1) [inline libmultilabel/data_utils.py:164]: how about the unk_init=False? not needed any more

File: libmultilabel/data_utils.py [hunk 4]
```diff
@@ -158,15 +160,21 @@ def load_or_build_text_dict(
...
     elif not embed_file.isdigit():
         logging.info(f'Load pretrained embedding from torchtext.')
         vocabs.load_vectors(embed_file, cache=embed_cache_dir)
     else:
         raise NotImplementedError
 
+    if normalize_embed:
+        embedding_weights = copy.deepcopy(vocabs.vectors.detach().cpu().numpy())
+        for i, vector in enumerate(embedding_weights):
+            embedding_weights[i] = vector/float(np.linalg.norm(vector) + 1e-6)
```
Comment (cjlin1) [inline libmultilabel/data_utils.py:174]: now I think my 1e-6 comment should be put here instead

File: libmultilabel/data_utils.py [hunk 5]
```diff
@@ -184,17 +192,14 @@ def load_or_build_label(datasets, label_file=None, silent=False):
     return classes
 
 
-def get_embedding_weights_from_file(word_dict, embed_file, silent=False, normalize=False):
+def get_embedding_weights_from_file(word_dict, embed_file, silent=False): # , normalize=False):
```
Comment (cjlin1) [inline libmultilabel/data_utils.py:195]:
```suggestion
def get_embedding_weights_from_file(word_dict, embed_file, silent=False):
```
I guess not need to mention normalize here..

File: libmultilabel/data_utils.py (hunk 3 above)
Comment (Eleven1Liu) [reply libmultilabel/data_utils.py:164]: According to the document (https://pytorch.org/text/0.9.0/_modules/torchtext/vocab.html#Vocab.set_vectors)
`unk_init` in `set_vectors` should be a callback function here. I'm not sure why we put `False` here.
For caml, removing `unk_init=False` won't the result. That's why I remove it.

**Review by cjlin1 — APPROVED** [review]
